src/sdk/namespace_cache.js

```diff
@@ -334,7 +337,9 @@ class NamespaceCache {
 
         let hub_read_stream;
         try {
-            hub_read_params.if_match_etag = params.object_md.etag;
+            if (_.isUndefined(hub_read_params.md_conditions)) {
+                hub_read_params.md_conditions = { if_match_etag: params.object_md.etag };
```

Comment:
Can you explain this logic? Is this the revalidation of an object with expired TTL? If so I would expect to send if_not_match instead so that hub will return an "error" if the object is the same instead of the data.

Reply:
I would also use if_modified in conjunction with it, for safer validation.

Reply:
opened issue https://github.com/noobaa/noobaa-core/issues/6146. resolving.